scripts/e2e-ci.sh

```diff
@@ -79,125 +82,140 @@ task() {
     task Model Listing Complete
 }
 
-test_serve() {
-    task Serve the model
-    local model_path
-    local serve_args
-
-    serve_args=()
-    model_path="${1:-}"
-    if [ -n "$model_path" ]; then
-        serve_args+=("--model-path" "${model_path}")
-    fi
-
-
-    # output serve log while `ilab model serve` is starting up
-    touch serve.log
-    tail -f serve.log &
-    TPID=$!
-
-    ilab model serve "${serve_args[@]}" &> serve.log &
-    wait_for_server start
-    kill "$TPID"
-    task Model Serving Complete
-}
-
-test_chat() {
-    task Chat with the model
-    CHAT_ARGS=("--endpoint-url" "http://localhost:8000/v1")
-    ilab model chat -m "${TRAINED_MODEL_PATH}" -qq "${CHAT_ARGS[@]}" 'Say "Hello" and nothing else\n'
-    task Chat with the model complete
-}
-
 test_taxonomy() {
     task Update the taxonomy
 
-    # add compositional skill to the taxomony
+    step Add compositional skill to the taxomony
     mkdir -p "$DATA_HOME"/instructlab/taxonomy/compositional_skills/extraction/inference/qualitative/e2e-siblings
     cp "$SCRIPTDIR"/test-data/e2e-qna-freeform-skill.yaml "$DATA_HOME"/instructlab/taxonomy/compositional_skills/extraction/inference/qualitative/e2e-siblings/qna.yaml
 
-    # add grounded skill to the taxomony
+    step Add grounded skill to the taxomony
     mkdir -p "$DATA_HOME"/instructlab/taxonomy/compositional_skills/extraction/answerability/e2e-yes_or_no
     cp "$SCRIPTDIR"/test-data/e2e-qna-grounded-skill.yaml "$DATA_HOME"/instructlab/taxonomy/compositional_skills/extraction/answerability/e2e-yes_or_no/qna.yaml
 
-    # add knowledge to the taxomony
+    step Add knowledge to the taxomony
     mkdir -p "$DATA_HOME"/instructlab/taxonomy/knowledge/phoenix/overview/e2e-phoenix
     cp "$SCRIPTDIR"/test-data/e2e-qna-knowledge-phoenix.yaml "$DATA_HOME"/instructlab/taxonomy/knowledge/phoenix/overview/e2e-phoenix/qna.yaml
 
-    step Taxonomy Verification
+    step Verify the taxonomy
     ilab taxonomy diff
-    task Taxonomy Updated and Verification complete
+
+    task Taxonomy updates and verification Complete
 }
 
 test_generate() {
+    # TODO
     task Generate synthetic data
-    ilab data generate
-    task Synthetic Data Generation Complete
+    local generate_args
+    generate_args=("--endpoint-url" "http://localhost:8000/v1" "--num-cpus" "$(nproc)" --taxonomy-path='./taxonomy') # needed?
+    if [ "$MEDIUM" -eq 1 ]; then
+        ilab data generate "${generate_args[@]}" --num-instructions 1
+    elif [ "$LARGE" -eq 1 ]; then
+        ilab data generate "${generate_args[@]}" --num-instructions 5
+    fi
+    task Synthetic data generation Complete
 }
 
 test_train() {
     task Train the model
+
     local knowledge_data_path
     knowledge_data_path=$(find "${DATA_HOME}"/instructlab/datasets -name 'knowledge_train_msgs*' | head -n 1)
-    ilab model train --data-path "${knowledge_data_path}"
+    if [ "$MEDIUM" -eq 1 ]; then
+        # TODO
+        ilab model train --pipeline=full \
```

Comment:
Each of the flags here should be set in the train profile or in `test_init` the way the large T-shirt size has been doing it. 

The only flag that should go into the command is the `--data-path`, which could also be set in the config as well, but since it's not known until after `ilab data generate` has been run, I think it's fine to use a flag there.

Reply:
Okay, when we talk let's just make sure all the correct values are being set - thanks for the info!